packages/pwa/app/utils/url.js

```diff
@@ -125,39 +142,42 @@ export const getUrlWithLocale = (shortCode, opts = {}) => {
             params.delete(param)
         })
     }
-
-    let paths = relativeUrl.split('/').filter((path) => path !== '')
-
-    if (localeParamType === urlPartPositions.PATH) {
-        // Array of the paths without empty items
-
-        // Remove the previous locale
-        paths.shift()
-
-        // Add the new locale
-        if (shortCode !== DEFAULT_LOCALE || paths?.length > 0) {
-            paths.unshift(shortCode)
+    if (relativeUrl === HOME_HREF) {
+        relativeUrl = buildPathWithUrlConfig(relativeUrl, {site: site?.alias, locale: shortCode})
+    } else {
+        let paths = relativeUrl.split('/').filter((path) => path !== '')
+        // remove the old locale and replace with new shortCode
+        if (localePosition === urlPartPositions.PATH && sitePosition === urlPartPositions.PATH) {
+            paths.splice(1, 1, shortCode)
+            console.log('paths', paths)
+        } else {
+            paths.splice(0, 1, shortCode)
         }
-    } else if (localeParamType === urlPartPositions.QUERY_PARAM) {
-        params.set('locale', shortCode)
-    }
-
-    relativeUrl = `/${paths.join('/')}${Array.from(params).length > 0 ? `?${params}` : ''}`
 
+        relativeUrl = `/${paths.join('/')}${Array.from(params).length > 0 ? `?${params}` : ''}`
+    }
     return relativeUrl
 }
 
 /**
- * Builds the Home page URL for a given locale.
- * We don't add the locale to the URL for the default locale.
+ * Builds the Home page URL for a given locale and site.
+ * locale and site wil be shown at all times except when both of them are default values
  *
+ * '/' default locale & site
+ * '/global/it-IT', '/?locale=it-IT&site=global' non-default locale, default site
+ * '/us', '/?site=us  default locale, non-default site
+ * '/us/it-IT' '/?locale=it-IT&site=us' non-default locale, non-default site
  * @param homeHref
- * @param locale
+ * @param options
  * @returns {string}
  */
-export const homeUrlBuilder = (homeHref, locale) => {
+export const homeUrlBuilder = (homeHref, options = {}) => {
+    const {locale, site} = options
+    const {l10n} = site
+    const defaultLocale = l10n.defaultLocale
     const updatedUrl = buildPathWithUrlConfig(homeHref, {
-        locale: locale !== DEFAULT_LOCALE ? locale : ''
+        locale: site?.id === DEFAULT_SITE_ID && locale === defaultLocale ? '' : locale,
```

Comment:
Looking at the code `DEFAULT_SITE_ID` is coming from the `package.json` file. We also have a default site id defined in the `pwa-app.config.json` file. I reckon that we should be referencing the later not the former.